src/resonate/dst/scheduler.py

```diff
@@ -374,6 +418,12 @@ def _resolve_promise(
             PromiseCompleted(promise_id=promise.promise_id, tick=self.tick, value=value)
         )
 
+    def _maybe_fail(self) -> bool:
```

Comment:
Moving failure simulation logic to a function. This will enable to simulate failure on more points on the system (before and after calling the durable promise storage)